packages/react-log-viewer/src/LogViewer/utils/utils.tsx

```diff
@@ -12,12 +12,12 @@ export const isArrayOfString = (array: string[]) => {
   It should take an array, and return an array of indexes where the searchedInput is found throughout the data array. 
   Should always be searching an array of strings. Look into lazy log for ideas.
 */
-export const searchForKeyword = (searchedInput: string, parsedData: string[]) => {
+export const searchForKeyword = (searchedInput: string, parsedData: string[], itemCount: number) => {
   const searchResults: number[] = [];
 
   parsedData.map((row, index) => {
     const regex = new RegExp(searchedInput, 'gi');
```

Comment:
I would probably take it out of the closure, I'm not sure if it gets optimized.

Reply:
I'm not sure what you mean by this

Reply:
iirc building the regex pattern takes some time, so if you use the same pattern it makes sense to have it shared across all iterations of the `map`, like the `searchResults`.

By optimalization I meant that with compiled languages the compiler can optimize it like I mentioned above.